lib/CMakeLists.txt

```diff
@@ -73,18 +65,38 @@ if (NOT TARGET picojson::picojson)
     find_package(picojson REQUIRED)
 endif()
 
-target_include_directories(mxl PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
-    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+target_include_directories(mxl-objects 
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
+)
+
+target_link_libraries(mxl-objects
+    PRIVATE
+        stduuid
+        spdlog::spdlog
+        fmt::fmt
+        picojson::picojson
+)
+
+add_library(mxl SHARED)
```

Comment:
Even though 1. `mxl` does not currently have any sources of it's own and 2. the feature is inherited from the `mxl-objects` dependency, I think it makes sense to keep the explicit mention of

```cmake
target_compile_features(mxl
    PRIVATE
        cxx_std_20
)
```

For future extensions where `mxl` might gain a few additional sources again, and to force an explicit clash if one target is updated without taking care of the others.